src/CodeGen_GPU_Dev.cpp

```diff
@@ -162,25 +167,56 @@ void CodeGen_GPU_C::visit(const Shuffle *op) {
         std::string src = vecs[0];
         std::ostringstream rhs;
         std::string storage_name = unique_name('_');
-        if (vector_declaration_style == VectorDeclarationStyle::OpenCLSyntax) {
+        switch (vector_declaration_style) {
+        case VectorDeclarationStyle::OpenCLSyntax:
             rhs << "(" << print_type(op->type) << ")(";
-        } else if (vector_declaration_style == VectorDeclarationStyle::WGSLSyntax) {
+            break;
+        case VectorDeclarationStyle::WGSLSyntax:
             rhs << print_type(op->type) << "(";
-        } else {
+            break;
+        case VectorDeclarationStyle::CLikeSyntax:
             rhs << "{";
+            break;
         }
+        int elem_num = 0;
         for (int i : op->indices) {
-            rhs << vecs[i];
-            if (i < (int)(op->indices.size() - 1)) {
+            size_t vector_idx;
```

Comment:
Comments needed for conditionals

Reply:
Added some comments, hope they are clear. 😄